### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by jlvFlores**
## For this PR, we:
- Added unit test for the methods inside each class  
- Added unit test for cases when the classes receive an incorrect amount of arguments

### Non-required changes
- Refactored the position of the files into corresponding folders and updated the `require './file_location'`
- Update the required arguments on student, teacher and person classes

### CODE REVIEW

File: spec/person_spec.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      expect(@person.can_use_services?).to eq(true)
+    end
+  end
+
+  context 'when initializing with invalid parameters' do
+    it 'raises an ArgumentError when not given three or four arguments' do
+      expect { Person.new(1, 'name') }.to raise_exception(ArgumentError)
+    end
+  end
+end
```
Comment (cynthiainga) [inline spec/person_spec.rb]: - [ ] Good job so far, but you just need to add some tests and you're good to go 👍🏾 
  - [x] You should add a test in the context when creating a new instance of the Person class:
    - A scenario of the constructor method returns a Person object with provided age, name, and empty rentals,
    - A scenario of the constructor method sets the default value for the optional name parameter if it is not provided
  - [x] You should add a test in the context when adding a new rentals:
    - A scenario of the add_rental method returns a rental and adds it to the book
  - [x] You should add a test in the context when checking permissions:
    - A scenario of the `can_use_services?` method returns false if under age and no parent permission,
    - A scenario of the `can_use_services?` method returns true if over age,
  - [x] You should add a test in the context when validating name:
    - A scenario of the correct_name method corrects the name, for example, if the name is `alongnameiscorrected` it should be equal to `Alongnamei`

**Review by cynthiainga — CHANGES_REQUESTED** [review]
Hi @jlvFlores & @elad237, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- [ ] All test files are placed in a different directory ✔️ 
- [ ] There is one file per class ✔️ 
- [ ] All implemented tests are passed so far ✔️ 

## Required Changes ♻️

- [ ] Please kindly check the comments under the review 👇🏾 👇🏾 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/person_spec.rb (hunk 1 above)
Comment (jlvFlores) [reply spec/person_spec.rb]: Hello @cynthiainga I've added the test you've asked except for this test, *A scenario of the constructor method sets the default value for the optional name parameter if it is not provided* as I've long changed the name parameter to be required. Hopefully you can understand the situation, I've even made sure to quickly and effectively add the other test to demonstrate our understanding of the topic, thank you for reading and have a nice day. 🙂

**Review by Shedrack-Sunday — APPROVED** [review]
Hi @jlvFlores 

Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉



Highlights
- [ ] Good code organization and  ✔️
- [ ] Descriptive README ✔️
- [ ] Clean test files 

Optional suggestions
Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**